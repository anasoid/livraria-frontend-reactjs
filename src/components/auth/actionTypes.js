export const AUTHENTICATE = 'AUTHENTICATE';
export const AUTH_ERROR = 'AUTH_ERROR';
export const SIGNOUT = 'SIGNOUT';