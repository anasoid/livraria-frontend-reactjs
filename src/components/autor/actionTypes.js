export const GET_AUTORES = 'GET_AUTORES';
export const ADD_AUTOR = 'ADD_AUTOR';
export const REMOVE_AUTOR = 'REMOVE_AUTOR';