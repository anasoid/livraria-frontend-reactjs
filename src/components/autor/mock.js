export const autores = {
  "autores": [
    {'nome': 'fulano', 'email': 'fulano@fulano.com'},
    {'nome': 'ciclano', 'email': 'ciclano@ciclano.com'}
  ]
};