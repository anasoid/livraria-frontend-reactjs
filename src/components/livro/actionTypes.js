export const GET_LIVROS = 'GET_LIVROS';
export const ADD_LIVRO = 'ADD_LIVRO';
export const REMOVE_LIVRO = 'REMOVE_LIVRO';
export const UPDATE_LIVRO = 'UPDATE_LIVRO';
export const SELECT_LIVRO = 'SELECT_LIVRO';