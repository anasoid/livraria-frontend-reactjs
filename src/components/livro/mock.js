export const livros = {
  livros: [
    {titulo: 'reactjs', isbn: '12345678', preco: 20.0, dtLancamento: '11/05/2018', autores: [{nome: 'fulano', email: 'fulano@email.com'}]},
    {titulo: 'react native', isbn: '87654321', preco: 15.0, dtLancamento: '20/08/2018', autores: [{nome: 'ciclano', email: 'ciclano@email.com'}]}
  ]
}