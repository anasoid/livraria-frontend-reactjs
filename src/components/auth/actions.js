import {AUTHENTICATE, AUTH_ERROR, SIGNOUT} from './actionTypes';
import {fakeAuth} from './fake-auth';

export const authenticate = (username, password) => {
  return dispatch => {
    return fakeAuth.authenticate(username, password)
      .then(() => {
        return dispatch({type: AUTHENTICATE, payload: { username, password }});
      }).catch( () => dispatch({type: AUTH_ERROR}));
  };
};

export const logout = () => {
  return dispatch => {
    return fakeAuth.signout()
      .then(() => {
        return dispatch({type: SIGNOUT});
      }).catch();
  };
};