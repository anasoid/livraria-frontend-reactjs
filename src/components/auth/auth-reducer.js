import {AUTH_ERROR, AUTHENTICATE, SIGNOUT} from './actionTypes';

const initialState = {
  isAuthenticated: false,
  authError: '',
  username: ''
};

const authReducer = (state = initialState, action) => {
  switch (action.type) {
    case AUTHENTICATE:
      return {
        ...state,
        isAuthenticated: true,
        username: action.payload.username
      };
    case AUTH_ERROR:
      return {
        ...state,
        isAuthenticated: false,
        authError: 'Usuario ou Senha incorretos'
      };
    case SIGNOUT:
      return initialState;
    default:
      return state;
  }
};

export default authReducer;