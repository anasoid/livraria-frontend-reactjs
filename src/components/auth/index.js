import * as actions from './actions';

export const authenticate = (username, password) => actions.authenticate(username, password);
export const logout = () => actions.logout();