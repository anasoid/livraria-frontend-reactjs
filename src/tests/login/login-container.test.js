import React from 'react';
import {MemoryRouter} from 'react-router';
import { Provider } from "react-redux";
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import LoginContainer from '../../components/login/login-container';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import initialState from '../util/initial-state';
import {AUTHENTICATE} from '../../components/auth/actionTypes';
import {createNodeMock} from '../util/mock';
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const expectedActions = [
  { type: AUTHENTICATE }
];

const store = mockStore(initialState, expectedActions);

describe('test LoginContainer component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <LoginContainer/>
        </MemoryRouter>
      </Provider>
    );
  });

  it('renders without crashing', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <LoginContainer/>
        </MemoryRouter>
      </Provider>,
      { createNodeMock }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('LoginContainer should be contain form', () => {
    expect(wrapper.find('form').at(0)).toExist();
  });

  it('LoginContainer should be contain input loginEmail', () => {
    const email = wrapper.find('#loginEmail').at(0);
    expect(email).toMatchSelector('TextField');
    expect(email).toHaveProp('name', 'email');
  });

  it('LoginContainer should be contain input loginSenha', () => {
    const senha = wrapper.find('#loginSenha').at(0);
    expect(senha).toMatchSelector('TextField');
    expect(senha).toHaveProp('name', 'senha');
  });

  it('LoginContainer should be contain button Efetue Login and tyoe submit', () => {
    const button = wrapper.find('button').at(0);
    expect(button).toHaveText('Efetue Login');
  });

});

describe('test submit form', () => {

  it('should submit form and handle prop function', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <LoginContainer/>
        </MemoryRouter>
      </Provider>
    );
    wrapper.find('form').at(0).simulate('submit', { target: { email: 'emailTest', senha: 'senhaTest' } });
  });
});

