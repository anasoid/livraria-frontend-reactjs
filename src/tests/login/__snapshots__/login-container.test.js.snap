// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LoginContainer component renders without crashing 1`] = `
<div>
  <div
    style={
      Object {
        "height": "auto",
        "left": 0,
        "margin": "auto",
        "maxWidth": 400,
        "minWidth": 320,
        "position": "absolute",
        "right": 0,
        "top": "20%",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "overflow": "auto",
          "padding": 20,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <span
        id="messages"
      >
        
      </span>
      <form
        onSubmit={[Function]}
      >
        <div
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        >
          <label
            htmlFor="loginEmail"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }
          >
            Email
          </label>
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 0,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            Email
          </div>
          <input
            disabled={false}
            id="loginEmail"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="email"
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        >
          <label
            htmlFor="loginSenha"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }
          >
            Senha
          </label>
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 0,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            Senha
          </div>
          <input
            disabled={false}
            id="loginSenha"
            name="senha"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitOpacity": 1,
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="password"
          />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottomStyle": "solid",
                  "borderBottomWidth": 2,
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "cursor": "pointer",
                "display": "table",
                "float": "left",
                "height": "auto",
                "maxWidth": 180,
                "muiPrepared": true,
                "overflow": "visible",
                "paddingTop": 5,
                "position": "relative",
                "width": "100%",
              }
            }
          >
            <input
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "cursor": "inherit",
                  "height": "100%",
                  "left": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 0,
                  "padding": 0,
                  "pointerEvents": "all",
                  "position": "absolute",
                  "width": "100%",
                  "zIndex": 2,
                }
              }
              type="checkbox"
            />
            <div
              style={
                Object {
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "borderColor": "#9e9e9e",
                    "color": "#9e9e9e",
                    "display": "block",
                    "fill": "#9e9e9e",
                    "flexShrink": 0,
                    "float": "left",
                    "height": 24,
                    "marginLeft": 0,
                    "marginRight": 16,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": 24,
                  }
                }
              >
                <div>
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "borderColor": "#9e9e9e",
                        "color": "#9e9e9e",
                        "display": "inline-block",
                        "fill": "#9e9e9e",
                        "height": 24,
                        "muiPrepared": true,
                        "opacity": 1,
                        "position": "absolute",
                        "transition": "opacity 1000ms cubic-bezier(0.23, 1, 0.32, 1) 200ms",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <svg
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "borderColor": "#9e9e9e",
                        "color": "#9e9e9e",
                        "display": "inline-block",
                        "fill": "#9e9e9e",
                        "height": 24,
                        "muiPrepared": true,
                        "opacity": 0,
                        "position": "absolute",
                        "transform": "scale(0)",
                        "transition": "opacity 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, transform 0ms cubic-bezier(0.23, 1, 0.32, 1) 450ms",
                        "transitionOrigin": "50% 50%",
                        "userSelect": "none",
                        "width": 24,
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </div>
                <div
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                />
                <div
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                />
              </div>
              <label
                style={
                  Object {
                    "color": "#9e9e9e",
                    "display": "block",
                    "float": "left",
                    "fontFamily": "Roboto, sans-serif",
                    "lineHeight": "24px",
                    "muiPrepared": true,
                    "position": "relative",
                    "width": "calc(100% - 60px)",
                  }
                }
              >
                Lembrar me
              </label>
            </div>
          </div>
          <div
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#ffffff",
                "borderRadius": 2,
                "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "float": "right",
                "fontFamily": "Roboto, sans-serif",
                "minWidth": 88,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <button
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "#00bcd4",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                  "width": "100%",
                }
              }
              tabIndex={0}
              type="submit"
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "backgroundColor": false,
                      "borderRadius": 2,
                      "height": 36,
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 14,
                        "fontWeight": 500,
                        "letterSpacing": 0,
                        "margin": 0,
                        "muiPrepared": true,
                        "opacity": 1,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "position": "relative",
                        "textTransform": "uppercase",
                        "userSelect": "none",
                      }
                    }
                  >
                    Efetue Login
                  </span>
                </div>
              </div>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
